File: php/testcode/securecookie_031.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_cookie_rememberme_predictable
function securecookie031(BenchmarkRequest $req): BenchmarkResponse {
    $userId = (int) $req->param('user_id');
    $token = $userId . ':' . date('YmdHis'); // vuln-code-snippet vuln-line php_cookie_rememberme_predictable
    setcookie('remember', $token, time() + 86400 * 30, '/', '', true, true);
    return BenchmarkResponse::ok('remember me set');
}
// vuln-code-snippet end php_cookie_rememberme_predictable

File: rust/testcode/securecookie_031.rs
//! CWE-614: Auth cookie constructed without HttpOnly or Secure flags despite short lifetime.

// vuln-code-snippet start testcodeSecurecookie031
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let token = req.param("token");
    let cookie = format!("auth={}; Max-Age=1", token); // vuln-code-snippet target-line testcodeSecurecookie031
    super::shared::BenchmarkResponse::ok(&format!("Set-Cookie: {}", cookie))
}
// vuln-code-snippet end testcodeSecurecookie031
